### ISSUE TITLE
No explanation of error returned from SharedInformer.AddEventHandler

### ISSUE THREAD

**Issue by MikeSpreitzer** [kubernetes/kubernetes#126235]
### What happened?

When reviewing controller code that reacts to an `error` from `SharedInformer.AddEventHandler`, I found myself unsure whether the code I was looking at is correct. The comments do not explain why the `AddEventHandler` method of `SharedInformer` can return an `error`. No mention of what errors can appear, what such an appearance would mean. The caller gets no clue about how to react to an error.

### What did you expect to happen?

I expected the interface to give enough information to write/evaluate calls to this method.

### How can we reproduce it (as minimally and precisely as possible)?

Look at the interface (e.g., https://github.com/kubernetes/client-go/blob/v0.26.0/tools/cache/shared_informer.go#L135-L140).

### Anything else we need to know?

Nope

### Kubernetes version

<details>

The `error` return has been there since release 1.26.0.

</details>

### Cloud provider

<details>
N/A
</details>

### OS version

<details>
N/A
</details>

### Install tools

<details>
N/A
</details>

### Container runtime (CRI) and version (if applicable)

<details>
N/A
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>
N/A
</details>

**Comment by cici37** [comment]
We are reviewing the issue in sig api-machinery bug triage meeting.
Hi @MikeSpreitzer, Thanks for the issue. Just curious if you plan to work on this or need help on this? Thanks

**Comment by MikeSpreitzer** [comment]
Yes, I could work on this. I do not yet actually know what errors can arise there. I will look into it and propose a comment.
